packages/ember-runtime/lib/system/core_object.js

```diff
@@ -540,7 +546,11 @@ CoreObject.PrototypeMixin = Mixin.create({
   toString() {
     let hasToStringExtension = typeof this.toStringExtension === 'function';
     let extension = hasToStringExtension ? `:${this.toStringExtension()}` : '';
-    let ret = `<${this[NAME_KEY] || this.constructor.toString()}:${guidFor(this)}${extension}>`;
+
+    let m = meta(this);
```

Comment:
550 and 551 are only required if `this[NAME_KEY]` isn't set, and it will be set most of the time. Can we avoid 550 and 551 unless we need to access it?

something like:

```js
this[NAME_KEY] || meta(this).getFactory() || this.cosntructor.toString()
```